x-pack/platform/plugins/private/upgrade_assistant/__jest__/client_integration/es_deprecations/es_deprecations.helpers.ts

```diff
@@ -95,9 +95,15 @@ const createActions = (testBed: TestBed) => {
     clickFilterByTitle: async (title: string) => {
       // We need to read the document "body" as the filter dropdown (an EuiSelectable)
       // is added in a portalled popover and not inside the component DOM tree.
-      const filterButton: HTMLButtonElement | null = document.body.querySelector(
-        `.euiSelectableListItem[title=${title}]`
-      );
+      let filterButton: HTMLButtonElement | null = null;
+
+      // Wait for the filterButton to be available
+      await act(async () => {
+        while (!filterButton) {
```

Comment:
This could potentially run indefinitely if the button never appears. Could we perhaps add a maxRetries here? there is also `waitFor` from `testing-library/react` which might be better suited, although not sure if we can use them here